be/src/storage/lake/starlet_location_provider.cpp

```diff
@@ -38,8 +41,23 @@ std::string StarletLocationProvider::join_path(std::string_view parent, std::str
                                 : fmt::format("{}/{}{}", prefix, child, suffix);
 }
 
-Status StarletLocationProvider::list_root_locations(std::set<std::string>*) const {
-    return Status::NotSupported("StarletLocationProvider::list_root_locations");
+Status StarletLocationProvider::list_root_locations(std::set<std::string>* roots) const {
+    if (roots == nullptr) {
+        return Status::InvalidArgument("roots set is NULL");
+    }
+    if (g_worker == nullptr) {
+        return Status::OK();
+    }
+    auto shards = g_worker->shards();
+    std::unordered_map<std::string, staros::starlet::ShardId> root_ids;
+    for (const auto& shard : shards) {
+        root_ids[shard.obj_store_info.s3_obj_store.uri] = shard.id;
```

Comment:
should check `shard.obj_store_info.scheme` first, make sure it is a `S3` type and then access `s3_obj_store`